src/components/MediaPlayer/VideoJSPlayer.js

```diff
@@ -116,66 +141,197 @@ function VideoJSPlayer({
         console.log('play');
         playerDispatch({ isPlaying: true, type: 'setPlayingStatus' });
       });
+      player.on('seeked', () => {
+        handleSeeked();
+      });
+      player.on('timeupdate', () => {
+        handleTimeUpdate();
+      });
     }
   }, [player]);
 
+  /**
+   * Switch canvas when using structure navigation / the media file ends
+   */
+  React.useEffect(() => {
+    if (isClicked && canvasIndex !== cIndex) {
+      switchPlayer();
+    }
+    setCIndex(canvasIndex);
+    setCanvasSegments(getSegmentMap({ manifest, canvasIndex }));
+  }, [canvasIndex]);
+
+  /**
+   * Update markers whenever player's currentTime is being
+   * updated. Time update happens when;
+   * 1. using structure navigation
+   * 2. seek and scrubbing events are fired
+   * 3. timeupdate event fired when playing the media file
+   */
   React.useEffect(() => {
     if (!player || !currentPlayer) {
       return;
     }
-
-    if (startTime != null) {
-      player.currentTime(startTime, playerDispatch({ type: 'resetClick' }));
-
-      // Mark current timefragment
+    if (currentNavItem != null && isReady) {
+      // Mark current time fragment
       if (player.markers) {
         player.markers.removeAll();
+        // Use currentNavItem's start and end time for marker creation
+        const { start, stop } = getMediaFragment(getItemId(currentNavItem));
+        playerDispatch({
+          endTime: stop,
+          startTime: start,
+          type: 'setTimeFragment',
+        });
         player.markers.add([
           {
-            time: startTime,
-            duration: endTime - startTime,
+            time: start,
+            duration: stop - start,
             text: currentNavItem.label.en[0],
           },
         ]);
       }
+    } else if (startTime == null) {
+      // When canvas gets loaded into the player, set the currentNavItem and startTime
+      // if there's a media fragment starting from time 0.0.
+      // This then triggers the creation of a fragment highlight in the player's timerail
+      const firstItem = getSegmentMap({ manifest, canvasIndex })[0];
+      const timeFragment = getMediaFragment(getItemId(firstItem));
+      if (timeFragment && timeFragment.start == 0) {
```

Comment:
Same here...

Reply:
Yes, I agree. It's safe to use strict equality in place of these. Especially in cases where, we are comparing objects. Good catch! I will change these.